tests/testthat/test_zm_range.R

```diff
@@ -1,83 +1,90 @@
-	expect_true( all( sf::st_z_range( pl ) == c(1, 4)))
-	expect_true( all( sf::st_z_range( pl ) == sf::st_z_range( pl ) ) )
-
-	mpl <- sf::st_multipolygon(x = list(pl, sf::st_polygon( x = list( matrix(c(0,0,10,10,0,-1,9,9,-1,-1,-10,-10,-1,0,-5,-5,0,0,10,10), ncol = 4, byrow = T) ) ) ) )
-	expect_true( all( sf::st_z_range( mpl ) == c(-10, 10) ) )
-	expect_true( all( sf::st_z_range( mpl ) == sf::st_z_range( mpl ) ) )
-
-	gc <- sf::st_geometrycollection(x = list(pt, mp))
-	expect_true( all( sf::st_z_range( gc ) == c(1, 5) ) )
-	expect_true( all( sf::st_z_range( gc ) == sf::st_z_range( gc ) ) )
-
-	gc <- sf::st_geometrycollection(x = list(ls, pl))
-	expect_true( all( sf::st_z_range( gc ) == c(1, 10) ) )
-	expect_true( all( sf::st_z_range( gc ) == sf::st_z_range( gc ) ) )
-
-	gc <- sf::st_geometrycollection(x = list(pt, mpl))
-	expect_true( all( sf::st_z_range( gc ) == c(-10, 10) ) )
-	expect_true( all( sf::st_z_range( gc ) == sf::st_z_range( gc ) ) )
-
+# Expect the z range, strip attributes to only compare values.
+expect_st_z_range <- function(object, expected) {
+	expect_equal(unclass(st_z_range(object)), expected, check.attributes = FALSE)
+}
+
+test_that("st_z_range and st_z_range returns correct value from sfg objects", {
+	
+	pt <- st_point(x = c(0,1,3,3))
+	
+	expect_st_z_range(pt, c(3,3))
+	expect_equal(st_z_range(pt), st_z_range(pt))
+
+	mp <- st_multipoint(x = matrix(c(0,1,1,1,0,2,5,5), ncol = 4, byrow = TRUE))
+	expect_st_z_range(mp, c(1, 5))
+	expect_equal(st_z_range(mp), st_z_range(mp))
+
+	ls <- st_linestring(x = matrix(c(0,1,1,1,0,2,5,5,0,3,10,10), ncol = 4, byrow = TRUE))
+	expect_st_z_range(ls, c(1, 10))
+	expect_equal(st_z_range(ls), st_z_range(ls))
+
+	mls <- st_multilinestring(x = list(ls, matrix(c(0,1,5,5,0,1,-1,-1), ncol = 4, byrow = TRUE)))
+	expect_st_z_range(mls, c(-1, 10))
+	expect_equal(st_z_range(mls), st_z_range(mls))
+
+	pl <- st_polygon(x = list(matrix(c(0,0,1,1,0,1,2,2,1,1,3,3,1,0,4,4,0,0,1,1), ncol = 4, byrow = T)))
+	expect_st_z_range(pl, c(1, 4))
+	expect_equal(st_z_range(pl), st_z_range(pl))
+
+	mpl <- st_multipolygon(
+		x = list(pl, st_polygon(
+			x = list(matrix(c(0,0,10,10,0,-1,9,9,-1,-1,-10,-10,-1,0,-5,-5,0,0,10,10),
+							 ncol = 4, byrow = TRUE))))
+		)
+	expect_st_z_range(mpl, c(-10, 10))
+	expect_equal(st_z_range(mpl), st_z_range(mpl))
+
+	gc <- st_geometrycollection(x = list(pt, mp))
+	expect_st_z_range(gc, c(1, 5))
+	expect_equal(st_z_range(gc), st_z_range(gc))
+
+	gc <- st_geometrycollection(x = list(ls, pl))
+	expect_st_z_range(gc, c(1, 10))
+	expect_equal(st_z_range(gc), st_z_range(gc))
+
+	gc <- st_geometrycollection(x = list(pt, mpl))
+	expect_st_z_range(gc, c(-10, 10))
+	expect_equal(st_z_range(gc), st_z_range(gc))
 })
 
 
 test_that("sf::st_z_range and sf::st_z_range returns correct value from sfc objects", {
+	pt <- st_sfc(st_point( x = c(0,1,3,3)))
+	# expect_equal(attr( pt, "zbox" ), c(3, 3)) # FIXME: now NULL
```

Comment:
Hi @edzer , please let me know if you agree with those changes, they are mostly stylistics, but when doing some linting, it helped me uncover this bug (or wrong test). If you want I don't mind breaking it down into smaller PRs.

The main goal is to standardize the tests, reduce noise before switching to testthat 3e.